frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx

```diff
@@ -38,12 +41,36 @@ const GitOpsListPage: React.FC = () => {
     };
   }, [baseURL, namespaces, nsError, nsLoaded]);
 
+  const [consoleLinks] = useK8sWatchResource({
+    isList: true,
+    kind: referenceForModel(ConsoleLinkModel),
+    optional: true,
+  });
+  let aLink;
+  _.filter(consoleLinks, (link: any) => {
+    if (link.spec.location === 'ApplicationMenu') {
+      if (link.metadata.name === "argocd") {
+        aLink = link;
+        return true;
+      }
+      return false;
+    }
+  });
```

Comment:
if I change this code snippet to your suggestion, I now get an error for `href={aLink.spec.href}` that says that `aLink` is of type `never` so therefore I can't get property `spec`. I've never worked with this type before, did you mean to change it to that on purpose?

Reply:
This is happening because you havent added `K8sResourceKind[]` as a type here https://github.com/openshift/console/pull/7127/files#diff-2091e68385783885a22919dac9db27a22a1eed67754375a6633cd0754d63e414R44 and by default the type is 'K8sResourceCommon[]' and since here the link type is `K8sResourceKind` and hence there is a type mismatch and the function never reaches an endpoint/or execute and no value is returned and thus the type of `aLink` becomes `never`. You can read more about `never` here https://www.tutorialsteacher.com/typescript/typescript-never